### DISCUSSION TITLE
Where to set the OPENAI-API-KEY

### DISCUSSION THREAD

**Discussion by jwhengeveld** [justLV/onju-voice#10] [Q&A]
After transcribing i keep getting this error

"2023-10-04 22:19:59,293 - ERROR - Attempt 1 of 4 failed: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://platform.openai.com/account/api-keys for details."

In what file do i need to set up the API key? I have tried credentials.json and config.yaml but it keeps this error

**Comment by justLV** [comment]
Set it as an environment variable as recommended by OpenAI

https://help.openai.com/en/articles/5112595-best-practices-for-api-key-safety

**Comment by jwhengeveld** [comment]
Thank you very much! This solved this error

Op di 10 okt 2023 04:49 schreef justLV ***@***.***>:
